### PR TITLE
[dist] [RFC] integrated backend tests in Travis-CI

### PR DESCRIPTION
**Pull request by lethliel**
This is the integration of the backend tests into Travis CI

- libdevel-cover-perl is needed
- the cleanup in test 0015 will not fail the test

@M0ses: Please discuss :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,20 @@
 language: ruby
+perl:
```
Comment (coolo) [inline .travis.yml]: this is ignored for ruby projects. So if you really want perl 5.18, you need to all perlbrew yourself - the default is 5.14

Comment (lethliel) [inline .travis.yml]: I was not aware of that. Thanks for the information.

Comment (coolo) [inline .travis.yml]: you can lint your travis.yml by using their command line client gem (packaged in dlre)

File: src/backend/t/0015-BSSched-EventSource-Directory.t [hunk 2]
```diff
@@ -257,7 +257,9 @@ push(@files_to_remove,"$ev_dir/$file_name");
 # CLEANUP
 unshift(@files_to_remove,"$eventdir_base/$arch/$evname");
 
-map { ( -d $_ ) ? rmdir $_ : unlink $_ ; die "$_: $!" if $!;} @files_to_remove;
+eval {
+  map { ( -d $_ ) ? rmdir $_ : unlink $_ ; die "$_: $!" if $!;} @files_to_remove;
```
Comment (M0ses) [inline src/backend/t/0015-BSSched-EventSource-Directory.t]: Why not just remove the "die" or change into a "warn" instead of "eval" the whole thing ?

**Comment by coolo** [comment]
BTW: for openqa I switched to codecov as it supports coverage for multiple language. Perhaps you should evaluate it too.

File: src/backend/t/0015-BSSched-EventSource-Directory.t (hunk 2 above)
Comment (lethliel) [inline src/backend/t/0015-BSSched-EventSource-Directory.t]: It was the first thing that came to my mind.

**Comment by lethliel** [comment]
> BTW: for openqa I switched to codecov as it supports coverage for multiple language. Perhaps you should evaluate it too.

@coolo looks nice. I will look at it.

File: src/backend/t/0015-BSSched-EventSource-Directory.t [hunk 3]
```diff
@@ -257,7 +257,7 @@ push(@files_to_remove,"$ev_dir/$file_name");
 # CLEANUP
 unshift(@files_to_remove,"$eventdir_base/$arch/$evname");
 
-map { ( -d $_ ) ? rmdir $_ : unlink $_ ; die "$_: $!" if $!;} @files_to_remove;
+map { ( -d $_ ) ? rmdir $_ : unlink $_ ; warn "$_: $!" if $!;} @files_to_remove;
```
Comment (mlschroe) [inline src/backend/t/0015-BSSched-EventSource-Directory.t]: Wow, now that's a horrible line. Please don't use map if you're really looking for a for() loop. And don't use multiple statements in a single line...
```
for (@files_to_remove) {
  multiple lines of your code here
}
```

Comment (lethliel) [inline src/backend/t/0015-BSSched-EventSource-Directory.t]: I changed it to a for loop.

**Pull request merged**